bigchaindb_driver/pool.py

```diff
@@ -46,9 +51,16 @@ def pick(self, connections):
                 :class:`~bigchaindb_driver.connection.Connection` instances.
 
         """
-        self.picked += 1
-        self.picked = self.picked % len(connections)
-        return connections[self.picked]
+
```

Comment:
In `pick` you only check for the node delay once - if the first connection is not available you return the second one without checking if it is available as well. This way at some point the driver will make requests to the previously failed nodes without respecting the delays.

Instead, `pick` should return the first available connection, somthing like this:

```
cursor = self.picked
while cursor < self.picked + len(connections):
    if current_time_ms > connections[cursor]['time]:
        self.picked = cursor // len(connections)
        return cursor
    
    cursor += 1
```

Reply:
i agree, so i updated accordingly with a minor change: 
```
   while current_time_ms <= connections[self.picked]['time']:
        self.next_node(connections)
        current_time_ms = datetime.utcnow()
   return connections[self.picked]
```
```next_node``` keeps iterating over complete set of nodes, until time during backoff is exhausted and we find an available node. please check last push.